src/ui/server/controllers/insurance/eligibility/buyer-country/index.test.ts

```diff
@@ -202,6 +202,41 @@ describe('controllers/insurance/eligibility/buyer-country', () => {
       });
     });
 
+    describe('when the submitted country does not have short term cover', () => {
+      const selectedCountryName = countryNoShortTermCover.isoCode;
+
+      beforeEach(() => {
+        req.body[FIELD_IDS.ELIGIBILITY.BUYER_COUNTRY] = selectedCountryName;
+
+        mockCountriesResponse = [countryNoShortTermCover];
+
+        getCisCountriesSpy = jest.fn(() => Promise.resolve(mockCountriesResponse));
+
+        api.keystone.APIM.getCisCountries = getCisCountriesSpy;
+      });
+
+      it('should update the session with populated country object', async () => {
+        await post(req, res);
+
+        const selectedCountry = getCountryByIsoCode(mockCountriesResponse, countryNoShortTermCover.isoCode) as Country;
+
+        const expectedPopulatedData = mapSubmittedEligibilityCountry(selectedCountry);
+
+        const expected = {
+          ...req.session.submittedData,
+          insuranceEligibility: updateSubmittedData(expectedPopulatedData, req.session.submittedData.insuranceEligibility),
+        };
+
+        expect(req.session.submittedData).toEqual(expected);
+      });
+
+      it(`should redirect to ${CONTRACT_TOO_SHORT}`, async () => {
+        await post(req, res);
+
+        expect(res.redirect).toHaveBeenCalledWith(CONTRACT_TOO_SHORT);
+      });
+    });
+
```

Comment:
Not a blocker, could do with splitting up this unit test into different files for different scenarios